### ISSUE TITLE
Speed up loading by using a single application instance

### ISSUE THREAD

**Issue by terrymacdonald** [terrymacdonald/DisplayMagician#73]
**Is your feature request related to a problem? Please describe.**
At the moment DisplayMagician takes a while to start any desktop shortcuts, or anytime the application is loaded from scratch. This is because at the moment when you doubleclick a desktop shortcut then a completely new instance of DisplayMagician loads and runs. This means that multiple copies of DisplayMagician can run at one time, and they use up memory, CPU time and cause delays while DisplayMagician loads.

**Describe the solution you'd like**
DisplayMagician should run as a single instance application, and the first instance should set up an IPC server that all subsequent instances communicate through. The idea is that when the user starts a game shortcut, the program will automatically realise it is the 2nd instance, and will send the command the user requested through to the first instance for processing. The 2nd instance will then close down as fast as possible.

The first instance will then perform the task that the user requested. 

The above architecture means that the user can set DisplayMagician to load on start up and to start minimised (which will happen when the PC is loading), after which time the interaction will be nice and quick. This will get around the need for the user to wait for DisplayMagician to scan all the games when it starts, as 2nd and subsequent shortcuts will simply reuse the already loaded DisplayMagician.

**Additional context**
Important things to note:
- The client code needs to be able to detect when the 1st DisplayMagician is busy running something, and report that error to the user before closing that additional instance.
- The server code needs to run on a separate thread so that it doesn't tie up the UI
- The server code needs to be able to run a game shortcut or a display profile change.
- The server code needs to shut down the server when the 1st instance is closed down.

**Comment by terrymacdonald** [comment]
Progress update. I've managed to get the command line passing to work so that any subsequent instances of DisplayMagician will actually pass their command line arguments to the first DisplayMagician, and then will close. The first DisplayMagician instance will load fully, and will service any requests from any subsequent DisplayMagician instances. It has to work this way to ensure that the Desktop shortcuts work, as each desktop shortcut will run a new instance of DisplayMagician, and the oly way to lock that down to a single instance is to use the singleinstance code I just developed.

So, that in turn brings some additional challenges. There is now the potential for a race condition, where a user doubleclicks a ChangeProfile shortcut at the same time as a RunShortcut profile, or some other combinations of those. Or, they may doubleclick a shortcut before the main application has fully started up. In either case, I need a way of getting the commandline processing to pause until the application has finished loading to avoid the race condition.

This is the next stage to be developed. I may also refactor the RunShortcut and RunProfile commands into background tasks so that I can stop locking the UI thread, but we'll have to see how that works out.

Thanks
Terry
